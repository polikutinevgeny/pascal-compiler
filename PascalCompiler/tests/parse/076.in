type
   testr = record
     i: Integer;
     j: Real;
   end;
   testa = array[3..4] of integer;
var
   a: integer;
   b: array[1..2] of integer = (0, 1);
   r: record
      i: Integer;
      j: Integer
   end = (i:1; j:2);
   c: testr = (i:1; j:2);
   d: testa = (1, 2);
begin
   write(a, b);
end.