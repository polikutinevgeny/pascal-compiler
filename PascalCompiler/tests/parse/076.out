└─Program ''
  └─Block
    ├─Declarations
    │ ├─Type section
    │ │ ├─testr
    │ │ │ └─Record
    │ │ │   └─Field list
    │ │ │     ├─Field declaration
    │ │ │     │ ├─integer
    │ │ │     │ └─i
    │ │ │     └─Field declaration
    │ │ │       ├─real
    │ │ │       └─j
    │ │ └─testa
    │ │   └─Array
    │ │     ├─integer
    │ │     └─Range
    │ │       ├─3
    │ │       └─4
    │ └─Var section
    │   ├─Variable declaration
    │   │ ├─integer
    │   │ └─a
    │   ├─Variable declaration
    │   │ ├─Array
    │   │ │ ├─integer
    │   │ │ └─Range
    │   │ │   ├─1
    │   │ │   └─2
    │   │ ├─b
    │   │ └─Array constant
    │   │   ├─0
    │   │   └─1
    │   ├─Variable declaration
    │   │ ├─Record
    │   │ │ └─Field list
    │   │ │   ├─Field declaration
    │   │ │   │ ├─integer
    │   │ │   │ └─i
    │   │ │   └─Field declaration
    │   │ │     ├─integer
    │   │ │     └─j
    │   │ ├─r
    │   │ └─Record constant
    │   │   ├─i
    │   │   │ └─1
    │   │   └─j
    │   │     └─2
    │   ├─Variable declaration
    │   │ ├─testr
    │   │ ├─c
    │   │ └─Record constant
    │   │   ├─i
    │   │   │ └─1
    │   │   └─j
    │   │     └─2
    │   └─Variable declaration
    │     ├─testa
    │     ├─d
    │     └─Array constant
    │       ├─1
    │       └─2
    └─Compound statement
      └─Statement list
        └─Write
          └─Expression list
            ├─Designator
            │ └─a
            └─Designator
              └─b
